app/models/subscription.rb

```diff
@@ -0,0 +1,15 @@
+class Subscription < ApplicationRecord
+  has_many :notifications, dependent: :restrict_with_error
```

Comment:
This implies that once we have a set of notifications (which hang around forever?) then we can't remove a subscription. But supplier has_many subscriptions with cascade destroy - so we can't remove a supplier either. IMHO this decision needs a least a supporting comment

Reply:
The notification records are like move records and shouldn't be hard-deleted...
...I've just see there is another PR to soft-delete moves, so will follow that pattern here